mayastor/src/grpc/mayastor_grpc.rs

```diff
@@ -32,14 +33,46 @@ use crate::{
 };
 use nix::errno::Errno;
 use rpc::mayastor::*;
-use std::{convert::TryFrom, ops::Deref};
+use std::{convert::TryFrom, ops::Deref, time::Duration};
 use tonic::{Request, Response, Status};
-
 #[derive(Debug)]
 struct UnixStream(tokio::net::UnixStream);
 
 #[derive(Debug)]
-pub struct MayastorSvc;
+pub struct MayastorSvc {
+    name: String,
+    interval: Duration,
+    rw_lock: tokio::sync::RwLock<bool>,
+}
+
+#[async_trait::async_trait]
+impl<F, R> Serializer<F, R> for MayastorSvc
+where
+    F: core::future::Future<Output = R> + Send + 'static,
+    R: Send,
+{
+    async fn locked(&self, f: F) -> R {
+        let mut lock = self.rw_lock.write().await;
+        trace!(?self.name, "locked");
+        assert!(!*lock);
+        *lock = true;
+        let output = f.await;
+        tokio::time::sleep(self.interval).await;
+        trace!(?self.name, "unlocked");
+        *lock = false;
+        output
+    }
+}
+
+impl MayastorSvc {
+    pub fn new(interval: Duration) -> Self {
+        Self {
+            name: String::from("CSISvc"),
```

Comment:
Should "CSISvc" be "MayastorSvc"?

Reply:
Yeah, maybe but as we have multiple services in "mayastor" calling mayastor SVC seems confusing to me?

Reply:
Fair enough :+1: